build.gradle

```diff
@@ -1,168 +1,64 @@
-// this part adds in ForgeGradle
 buildscript {
     repositories {
-        mavenCentral()
-        maven {
-            name = "forge"
-            url = "http://files.minecraftforge.net/maven"
-        }
-        maven {
-            name = "sonatype"
-            url = "https://oss.sonatype.org/content/repositories/snapshots/"
-        }
+        jcenter()
+        maven { url = "http://files.minecraftforge.net/maven" }
+    }
+    dependencies {
+        classpath 'net.minecraftforge.gradle:ForgeGradle:2.3-SNAPSHOT'
     }
-    dependencies { classpath 'net.minecraftforge.gradle:ForgeGradle:1.2-SNAPSHOT' }
 }
+apply plugin: 'net.minecraftforge.gradle.forge'
+//Only edit below this line, the above code adds and enables the necessary things for Forge to be setup.
+
 
-apply plugin: "forge"
+version = "2.0.0"
+group = "com.wynprice.secretroomsmod" // http://maven.apache.org/guides/mini/guide-naming-conventions.html
+archivesBaseName = "secretroomsmod"
 
-// stuff specific to my mod
-group = 'com.github.abrarsyed'
-version = "4.7.1"
-archivesBaseName = 'secretroomsmod'
+sourceCompatibility = targetCompatibility = '1.8' // Need this here so eclipse task generates correctly.
+compileJava {
+    sourceCompatibility = targetCompatibility = '1.8'
+}
 
 minecraft {
-    version = "1.7.10-10.13.2.1291" // grab latest forge
+    version = "1.12.2-14.23.0.2512"
     runDir = "run"
-
-    // replacing
-    replace "@CHANGE_VERSION@", "[4.7,)"
-    replace "@MC_VERSION@", project.minecraft.version
-    replace "@VERSION@", project.version
-    replaceIn "SecretRooms.java"
```

Comment:
You should probably change these values, but not remove them entirely. I might have been out of the community for quite a while, but Im still pretty sure this info is useful to keep in the `@Mod`.